src/banded/bidiagonalconjugationdata.jl

```diff
@@ -0,0 +1,85 @@
+"""
+    BidiagonalConjugationData{T, MU, MC} <: LazyMatrix{T}
+
+Struct for efficiently representing the matrix product `A = inv(U)XV`,
+assuming that 
+
+- `A` is upper bidiagonal, 
+- `U` is upper Hessenberg,
+- `X` is banded,
+- `V` is banded.
+
+None of these properties are checked internally. It is the user's responsibility
+to ensure these properties hold and that the product `inv(U)XV` is indeed bidiagonal.
```

Comment:
Does our algorithm change if we just think of this as a bidiagonal projection of a possibly dense matrix? If not we don't have to include this caveat.

Reply:
It doesn't change, so yeah I can remove that note.

Reply:
If this computes bidiagonal projections in the general case, are you saying if I just wrap this in `Symmetric(...)` then it would work for the orthonormal symmetric tridiagonal Jacobi matrices given connection coefficients in `U` and `V`?

Reply:
@TSGut I'm not exactly familiar with what you mean. Can you give an example?